analytics.php

```diff
@@ -0,0 +1,231 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Page to grant external users access to a BBB session
+ *
+ * @package    mod_bigbluebuttonbn
+ * @copyright  Kevin Pham <kevinpham@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+use mod_bigbluebuttonbn\plugin;
+
+require(__DIR__.'/../../config.php');
+require_once(__DIR__.'/locallib.php');
+require_once(__DIR__.'/viewlib.php');
+
+$id = required_param('id', PARAM_RAW);
+$bn = optional_param('bn', 0, PARAM_INT);
+$group = optional_param('group', 0, PARAM_INT);
+
+$viewinstance = bigbluebuttonbn_view_validator(null, $bn); // In locallib.
+if (!$viewinstance) {
+    print_error('view_error_url_missing_parameters', plugin::COMPONENT);
```

Comment:
Can this be a throw new moodle_exception instead?

Reply:
I looked it up, it does throw a moodle exception under the hood @ lib/setuplib.php:498. Did you want me to replace this with a direct moodle exception throw instead?

Reply:
If you wanted to throw a more specific exception, eg invalid_parameter_exception